src/checkers/inference/model/serialization/CnfVecIntSerializer.java

```diff
@@ -151,7 +151,10 @@ public CnfVecIntSerializer(SlotManager slotManager) {
         //TODO: WE SHOULD INSTEAD PIPE THROUGH THE ExistentialVariable ID
         Integer existentialId = existentialToPotentialVar.get(constraint.getPotentialVariable().getId());
         if (existentialId == null) {
-            existentialId = slotManager.nextId();
+            // existentialId should not overlap with the Id of real slots in slot manager
+            // thus by computing sum of total slots number in slot manager
+            // and the size of existentialToPotentialVar and plus 1 to get next id of existential Id here
+            existentialId = slotManager.getNumberOfSlots() + existentialToPotentialVar.size() + 1;
```

Comment:
I'm wondering whether the result of `getNumberOfSlots` needs to be multiplied with the number of bits used for each slots. Is the existentialId used as a bit for the SAT solver? Or will it be multiplied later?